packages/multi-test/src/staking.rs

```diff
@@ -15,12 +29,240 @@ pub enum StakingSudo {
+    }
+
+    fn query(
+        &self,
+        api: &dyn Api,
+        storage: &dyn Storage,
+        _querier: &dyn Querier,
+        _block: &BlockInfo,
+        request: StakingQuery,
+    ) -> AnyResult<Binary> {
+        let staking_storage = prefixed_read(storage, NAMESPACE_STAKING);
+        match request {
+            StakingQuery::Delegation {
+                delegator,
+                validator,
+            } => {
+                // for now validator is ignored, as I want to support only one validator
+                let delegator = api.addr_validate(&delegator)?;
+                let stakes = match self.get_stakes(&staking_storage, &delegator) {
+                    Ok(stakes) => stakes[0].clone(),
+                    Err(_) => {
+                        let response = DelegationResponse { delegation: None };
+                        return Ok(to_binary(&response)?);
+                    }
+                };
+                // set fixed reward ratio 1:10 per delegated amoutn
+                let reward = coin(
+                    (stakes.amount / Uint128::new(10)).u128(),
+                    stakes.denom.clone(),
+                );
+                let full_delegation_response = DelegationResponse {
+                    delegation: Some(FullDelegation {
+                        delegator,
+                        validator,
+                        amount: stakes,
+                        can_redelegate: coin(0, "testcoin"),
+                        accumulated_rewards: vec![reward],
+                    }),
+                };
+
+                let res = to_binary(&full_delegation_response)?;
+                Ok(res)
+            }
+            q => bail!("Unsupported staking sudo message: {:?}", q),
+        }
+    }
+}
+
+#[derive(Default)]
+pub struct DistributionKeeper {}
+
+impl DistributionKeeper {
+    pub fn new() -> Self {
+        DistributionKeeper {}
+    }
+
+    fn get_stakes(&self, storage: &dyn Storage, account: &Addr) -> AnyResult<Vec<Coin>> {
+        let val = STAKES.may_load(storage, account)?;
+        Ok(val.unwrap_or_default().into_vec())
+    }
+}
+
+impl Distribution for DistributionKeeper {}
+
+impl Module for DistributionKeeper {
+    type ExecT = DistributionMsg;
+    type QueryT = Empty;
+    type SudoT = Empty;
+
+    fn execute<ExecC, QueryC>(
+        &self,
+        _api: &dyn Api,
+        storage: &mut dyn Storage,
+        _router: &dyn CosmosRouter<ExecC = ExecC, QueryC = QueryC>,
+        _block: &BlockInfo,
+        sender: Addr,
+        msg: DistributionMsg,
+    ) -> AnyResult<AppResponse> {
+        let staking_storage = prefixed(storage, NAMESPACE_STAKING);
+        match msg {
+            // For now it ignores validator as I want to support only one
+            DistributionMsg::WithdrawDelegatorReward { validator } => {
+                let stakes = self.get_stakes(&staking_storage, &sender)?[0].clone();
+                // set fixed reward ratio 1:10 per delegated amoutn
```

Comment:
As above in the staking query, need a good way to configure how many rewards.
Maybe we do it in this module and the staking module calls this internally to service this request?